src/Components/Table/types.ts

```diff
@@ -0,0 +1,24 @@
+import { SxProps } from "@mui/material";
+import { MouseEvent } from "react";
+import { GridRenderCellParams, GridSortModel } from "@mui/x-data-grid";
+import { sortType } from "./consts";
+export type TableCol = {
+  title: string;
+  field: string;
+  width?: number;
+  flex?: number;
+  minWidth?: number;
+  renderCell?: (params: GridRenderCellParams) => React.ReactNode;
+};
+
+export type TableProps = {
+  cols: TableCol[];
+  rows: any[];
+  onRowClick: (event: MouseEvent<HTMLElement>) => void;
+  style?: SxProps;
+  headerClassName?: string;
+  headerTypo: (label: string) => React.ReactNode;
+  disableColumnSorting?: boolean;
+  disableColumnMenu?: boolean;
```

Comment:
what is this prop for?

Reply:
In data grid of mui we got a sort and filter for each column automatically (it comes as a drop down menu) . This prop is to disable them.